spec.md

```diff
@@ -0,0 +1,948 @@
+> They simply indicate that a song should be started or stopped a certain amount of time into the audio file.
+
+> [!WARNING]
+>
+> **Breaking Change** in version 2.0.0
+>
+> In versions 0.x and 1.x `START` was specified in seconds as a floating point number.
+> Version 2.0.0 changes this to an integer and milliseconds.
+
+### 3.13. The `PREVIEWSTART` Header
+
+```
+Required:     No
+Multi-Valued: No
+Syntax:       1*DIGIT
+Since:        0.2.0
+```
+
+The `PREVIEWSTART` header indicates a time offset in milliseconds relative to the start of the audio where the preview starts.
+Implementations MAY use this value when playing a song in a preview setting (e.g. during song selection).
+In its absence implementations SHOULD default to the start of the medley section (if available).
+
+> [!WARNING]
+>
+> **Breaking Change** in version 2.0.0
+>
+> In versions 0.x and 1.x `PREVIEWSTART` was specified in seconds as a floating point number.
+> Version 2.0.0 of this specification changes this to an integer and milliseconds.
+
+### 3.14. The `MEDLEYSTART` and `MEDLEYEND` Headers
+
+```
+Required:     No
+Multi-Valued: No
+Syntax:       1*DIGIT
+Since:        2.0.0
+```
+
+The `MEDLEYSTART` and `MEDLEYEND` headers indicate in milliseconds the start and end of the medley section of a song relative to the start of the audio.
+These tags replace `MEDLEYSTARTBEAT` and `MEDLEYENDBEAT` in version 2.0.0 of this specification.
+
+### 3.15. The `MEDLEYSTARTBEAT` and `MEDLEYENDBEAT` Headers
+
+```
+Required:     No
+Multi-Valued: No
+Syntax:       1*DIGIT
+Since:        0.2.0
+Removed:      2.0.0
+```
+
+The `MEDLEYSTARTBEAT` and `MEDLEYENDBEAT` headers indicate in beats the start and end of the medley section of a song.
+Implementations MUST respect the `GAP` value when calculating the medley start and end times.
+
+> [!WARNING]
+>
+> **Breaking Change** in version 2.0.0
+>
+> The headers `MEDLEYSTARTBEAT` and `MEDLEYENDBEAT` have been replaced by `MEDLEYSTART` and `MEDLEYEND` in version 2.0.0 of this specification.
+
+### 3.16. The `CALCMEDLEY` Header
+
+```
+Required:     No
+Multi-Valued: No
+Syntax:       "on" / "off"
+Since:        0.2.0
+```
+
+If `MEDLEYSTART` or `MEDLEYEND` (`MEDLEYSTARTBEAT` or `MEDLEYENDBEAT` for versions before 2.0.0) are not specified,
+the `CALCMEDLEY` header indicates whether an implementation is supposed to determine the start and end of the medley section automatically.
+The value of this header is compared case-insensitively.
+
+> [!CAUTION]
+>
+> The exact semantics of the `CALCMEDLEY` header have not been defined yet.
+
+### 3.17. The `TITLE` Header
+
+```
+Required:     Yes
+Multi-Valued: No
+Since:        0.1.0
+```
+
+The `TITLE` header contains the title of the song.
+
+### 3.18. The `ARTIST` Header
+
+```
+Required:     Yes
+Multi-Valued: No
+Since:        0.1.0
+```
+
+The `ARTIST` header contains the artist of the song.
+
+### 3.19. The `YEAR` Header
+
+```
+Required:     No
+Multi-Valued: No
+Syntax:       4DIGIT
```

Comment:
For my taste it could even be `1*DIGIT` or even `[ minus ] 1*DIGIT`. I chose `4DIGIT` because the [website](https://usdx.eu/format/#year) currently says the year should be a 4-digit year.

I updated this to `4*DIGIT` and will propose the other options in a follow-up issue.